### PR TITLE
In progress: PORTALS-368

### PR DESCRIPTION
**Pull request by leem42**

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -10,6 +10,6 @@
   "parserOptions": {
     "ecmaFeatures": {
       "jsx": true
-    },
```
Comment (leem42) [inline .eslintrc.json]: fixes broken vscode intellisense

File: README.md [hunk 2]
```diff
@@ -224,7 +224,7 @@ An example of a view with facets/stacked bar chart/table
           }}
           rgbIndex={0}
           token={this.state.token}
-          filter={"assay"}
```
Comment (leem42) [inline README.md]: better naming, its what is used on the backend

File: package.json [hunk 3]
```diff
@@ -58,14 +58,14 @@
...
     "react-measure": "^2.1.2",
-    "react-native-rss-parser": "^1.3.0",
     "react-plotly.js": "^2.3.0",
     "react-router-dom": "^4.3.1",
     "react-scripts": "^3.0.1",
     "react-test-renderer": "^16.4.2",
     "react-tooltip": "^3.9.2",
     "react-transition-group": "2.6.0",
     "rollup-plugin-babel-minify": "^8.0.0",
+    "rss-parser": "^3.7.2",
```
Comment (leem42) [inline package.json]: has umd support

File: src/__tests__/lib/containers/GenericCard.test.tsx [hunk 4]
```diff
@@ -108,6 +108,11 @@ describe('it renders the UI correctly', () => {
     expect(wrapper.find(Utils.CardFooter)).toHaveLength(1)
   })
 
+  it.skip('renders the query handles corretly', () => {
```
Comment (leem42) [inline src/__tests__/lib/containers/GenericCard.test.tsx]: need a test for validating data attributes, enzyme isn't rendering data attributes though

File: src/__tests__/lib/containers/QueryWrapperMenu.test.tsx [hunk 5]
```diff
@@ -170,4 +205,77 @@ describe('it renders an accordion config', () => {
     expect(wrapper.find(ACCORDION_GROUP_MENU_SELECTOR).childAt(2).hasClass('SRC-pointed-triangle-right')).toBeTruthy()
   })
 
+  it('passes down the correct unitDescription correctly' , async () => {
```
Comment (leem42) [inline src/__tests__/lib/containers/QueryWrapperMenu.test.tsx]: validates the hard coded label were using currently

File: src/__tests__/lib/containers/QueryWrapperMenu.test.tsx [hunk 6]
```diff
@@ -170,4 +205,77 @@ describe('it renders an accordion config', () => {
...
+    rgbIndex: 3,
+  }
+
+  it('renders correctly and passes down the right unitDescription', async () => {
+    const { wrapper } = await createMountedComponent(props)
+    const queryWrapperWithSearch = wrapper.find(QueryWrapper).at(2)
+    expect(queryWrapperWithSearch.props().unitDescription).toEqual('Computational Tools')
+    // Search only makes sense with an unfaceted view, so there's additional logic to make sure only
+    // in the case of search that facets are excluded in QueryWrapperMenu
+    expect(queryWrapperWithSearch.find(Facets)).toHaveLength(0)
```
Comment (leem42) [inline src/__tests__/lib/containers/QueryWrapperMenu.test.tsx]: this is the only configuration when facets aren't shown, we also have to test unitDescription is shown correctly

File: src/__tests__/lib/utils/modules/urlConfiguration.test.ts [hunk 7]
```diff
@@ -1,12 +1,12 @@
-import { insertWhereClauseFromURL } from '../../../../lib/utils/modules/sqlFunctions'
+import { insertWhereClauseFromSearchParams } from '../../../../lib/utils/modules/sqlFunctions'
```
Comment (leem42) [inline src/__tests__/lib/utils/modules/urlConfiguration.test.ts]: more generic name

File: package.json (hunk 3 above)
Comment (jay-hodgson) [reply package.json]: please try to exclude instead (this other rss parser does not provide access to the full html content)

File: src/lib/containers/QueryWrapper.tsx [hunk 8]
```diff
@@ -51,12 +52,13 @@ export type QueryWrapperChildProps = {
   isLoading?: boolean
   token?: string
   isLoadingNewData?: boolean
-  executeQueryRequest?: (param: QueryBundleRequest) => void
+  executeQueryRequest?: (param: QueryBundleRequest) => Promise<undefined>
```
Comment (leem42) [inline src/lib/containers/QueryWrapper.tsx]: dont need

File: src/lib/containers/QueryWrapper.tsx [hunk 9]
```diff
@@ -175,9 +189,11 @@ export default class QueryWrapper extends React.Component<QueryWrapperProps, Que
           asyncJobStatus: undefined
         }
         this.setState(newState)
+        return Promise.resolve(() => {})
       }
     ).catch((err: string) => {
       console.log('Failed to get data ', err)
+      return Promise.resolve(() => {})
```
Comment (leem42) [inline src/lib/containers/QueryWrapper.tsx]: not needed

File: src/lib/containers/RssFeed.tsx [hunk 10]
```diff
@@ -1,5 +1,6 @@
 import * as React from 'react'
-import reactNativeRssParser from 'react-native-rss-parser'
+let Parser = require('rss-parser')
```
Comment (leem42) [inline src/lib/containers/RssFeed.tsx]: revert

File: src/lib/style/Portal.scss [hunk 11]
```diff
@@ -556,7 +556,7 @@ a.SRC-anchor-light {
   background: unset;
 }
 .SRC-menu-button-base:focus {
-  outline: none;
+  outline: $primaryColor;
 }
```
Comment (leem42) [inline src/lib/style/Portal.scss]: export file and import into portals repo!

File: src/lib/style/Portal.scss [hunk 12]
```diff
@@ -1363,3 +1363,30 @@ button {
     color:white;
   }
 }
+
+.highlight {
+  // background: rgb(244, 248, 253);
```
Comment (leem42) [inline src/lib/style/Portal.scss]: remove

File: yarn.lock [hunk 13]
```diff
@@ -9025,7 +9020,7 @@ react-measure@^2.1.2:
     prop-types "^15.6.2"
     resize-observer-polyfill "^1.5.0"
 
-react-native-rss-parser@^1.3.0:
+react-native-rss-parser@1.4.0:
```
Comment (jay-hodgson) [inline yarn.lock]: seems strange that react-native-rss-parser is still in the yarn.lock

**Pull request merged**